### DISCUSSION TITLE
How can I transmit room temperature to BSB-LAN over MQTT in JSON format?

### DISCUSSION THREAD

**Discussion by HiperiorNr1** [fredlcore/BSB-LAN#612] [Q&A] [answered]
Unfortunately, for some reason, BSB-LAN via MQTT will only work with JSON. This brings up a few problems to solve, including how can I set parameters or send temperatures via MQTT? 
This is not documented, only the plain text version.
I tried to figure it out from the plain text description and examples, but it refuses to work that way.

An example I receive via MQTT:
BSB-LAN/json: {"BaxiPlatinum":{"status":{"8771":"23.5"}}}

For example for the temperature to be sent:
BSB-LAN/json: {"BaxiPlatinum":{"status":{"I10000":"21.3"}}}

If I send it via http, it works:
http://172.19.253.8/I10000=21.3

Should it be posted in another topic, or is there another way to do it?
The settings don't work this way either.

It's worth a separate question here as to why bsb-lan doesn't work in plain text format, but if I've got this far, I'll fight my way through it :)

**Comment by fredlcore** [comment, marked as answer]
BSB-LAN via MQTT also supports JSON, but only in the direction BSB-LAN -> MQTT broker, i.e. for reading from parameters.
In order to send (set) a parameter via MQTT to BSB-LAN, the MQTT payload has to be something like `I10000=21.3`. Sending payload in JSON format is not implemented. If someone contributes the code for this, I'm happy to add it, but since I myself do not use MQTT, I won't be the one to do it.
